src/libopensc/card-openpgp.c

```diff
@@ -2933,6 +2946,10 @@ pgp_parse_and_set_pubkey_output(sc_card_t *card, u8* data, size_t data_len,
 		}
 		/* ECC public key */
 		else if (tag == 0x0086) {
+			if (key_info->algorithm != SC_OPENPGP_KEYALGO_ECDSA &&
+					key_info->algorithm != SC_OPENPGP_KEYALGO_ECDH) {
+				LOG_FUNC_RETURN(card->ctx, SC_ERROR_UNKNOWN_DATA_RECEIVED);
+			}
 			/* set the output data */
 			/* len is ecpoint length + format byte
 			 * see section 7.2.14 of 3.3.1 specs */
```

Comment:
Shouldn't also lines below `key_info->u.ec.ecpoint = malloc(len);` set the `key_info->u.ec.ecpoint_len` to the `len`?

Reply:
Good point.  In `src/pkcs15init/pkcs15-openpgp.c` where the overrun happened, the `ecpoint_len` is coming in the `required->field_length` and that one comes from the algorithm/mechanism specification.

This code on the following lines checks if the length is different than expected and I think it should really fail if the length is not the expected one (2z + 1) from specs in https://gnupg.org/ftp/specs/OpenPGP-smart-card-application-3.4.pdf

Reply:
Having further look in the processing, you are right and the length should be set here otherwise we will overrun the buffers somewhere later.